Source/NSSplitViewController.m

```diff
@@ -0,0 +1,196 @@
+}
+
+- (void) setSplitView: (NSSplitView *)splitView
+{
+  [self setView: splitView];
+  [splitView setDelegate: self];
+}
+
+- (NSSplitViewItem *) splitViewItemForViewController: (NSViewController *)vc
+{
+  FOR_IN (NSSplitViewItem*, svi, _splitViewItems)
+    if ([svi viewController] == vc)
+      {
+        return svi;
+      }
+  END_FOR_IN (_splitViewItems);
+  return nil;
+}
+
+- (CGFloat) minimumThicknessForInlineSidebars
+{
+  return _minimumThicknessForInlineSidebars;
+}
+  
+// manage splitview items...
+- (NSArray *) splitViewItems
+{
+  return _splitViewItems;
+}
+
+- (void) addSplitViewItem: (NSSplitViewItem *)item
+{
+  NSViewController *vc = [item viewController];
+  [_splitViewItems addObject: item];
+  if(vc != nil)
+    {
+      NSView *v = [vc view];
+      if (v != nil)
+        {
+          [[self splitView] addSubview: v];
+        }
+    }
+}
+
+- (void) insertSplitViewItem: (NSSplitViewItem *)item atIndex: (NSInteger)index
+{
+  NSSplitView *sv = [self splitView];
+  NSViewController *vc = [item viewController];
+
+  if (vc != nil)
+    {
+      NSView *v = [vc view];
+      if (v != nil)
+        {
+          [sv insertSubview: v atIndex: index];
+        }
+    }       
+  [_splitViewItems insertObject: item atIndex: index];     
+}
+
+- (void) removeSplitViewItem: (NSSplitViewItem *)item
+{
+  NSViewController *vc = [item viewController];
+  if (vc != nil)
+    {
+      NSView *v = [vc view];
+      if (v != nil)
+        {
+          [[self splitView] removeSubview: v];  
+        }
+    }
+  [_splitViewItems removeObject: item];
+}
+
+// instance methods...
+- (NSRect)splitView:(NSSplitView *)splitView additionalEffectiveRectOfDividerAtIndex:(NSInteger)dividerIndex
+{
+  return [splitView frame];
+}
+
+- (BOOL)splitView:(NSSplitView *)splitView canCollapseSubview:(NSView *)subview
+{
+  return YES;
+}
+
+- (NSRect)splitView:(NSSplitView *)splitView effectiveRect:(NSRect)proposedEffectiveRect forDrawnRect:(NSRect)drawnRect
+   ofDividerAtIndex:(NSInteger)dividerIndex
+{
+  return proposedEffectiveRect;
+}
+
+- (BOOL)splitView:(NSSplitView *)splitView shouldCollapseSubview:(NSView *)subview forDoubleClickOnDividerAtIndex:(NSInteger)dividerIndex
+{
+  return YES;
+}
+
+- (BOOL)splitView:(NSSplitView *)splitView shouldHideDividerAtIndex:(NSInteger)dividerIndex
+{
+  return YES;
+}
+
+- (IBAction)toggleSidebar:(id)sender
+{
+  NSLog(@"Toggle");
+}
+
+// NSCoding
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+  if ([coder allowsKeyedCoding])
+    {
+      if ([coder containsValueForKey: @"NSSplitView"])
+        {
+          NSSplitView *sv = [coder decodeObjectForKey: @"NSSplitView"];
+          [self setSplitView: sv];
+        }
+      if ([coder containsValueForKey: @"NSSplitViewItems"])
+        {
+          NSArray *items = [coder decodeObjectForKey: @"NSSplitViewItems"];
+          [_splitViewItems addObjectsFromArray: items];
+        }
+    }
```

Comment:
We always should add the non-keyed code as well.

Reply:
Yes, I am trying to complete the keyed stuff first, it needs work

Reply:
Done